nav2_route/src/graph_loader.cpp

```diff
@@ -0,0 +1,127 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+
+#include "nav2_route/graph_loader.hpp"
+#include "ament_index_cpp/get_package_share_directory.hpp"
+
+namespace nav2_route
+{
+
+GraphLoader::GraphLoader(
+  nav2_util::LifecycleNode::SharedPtr node,
+  std::shared_ptr<tf2_ros::Buffer> tf,
+  const std::string frame)
+: plugin_loader_("nav2_route", "nav2_route::GraphFileLoader"),
+  default_plugin_id_("GeoJsonGraphFileLoader")
+{
+  logger_ = node->get_logger();
+  tf_ = tf;
+  route_frame_ = frame;
+
+  if (!node->has_parameter("graph_filepath")) {
+    throw std::runtime_error("The parameter graph_filepath was not defined");
+  }
+
+  graph_filepath_ = node->get_parameter("graph_filepath").as_string();
+
+  // Default Graph Parser
+  const std::string default_plugin_type = "nav2_route::GeoJsonGraphFileLoader";
+
+  nav2_util::declare_parameter_if_not_declared(
+    node, "graph_file_loader", rclcpp::ParameterValue(default_plugin_id_));
+  auto graph_file_loader_id = node->get_parameter("graph_file_loader").as_string();
+
+  if (graph_file_loader_id == default_plugin_id_) {
+    nav2_util::declare_parameter_if_not_declared(
+      node, default_plugin_id_ + ".plugin", rclcpp::ParameterValue(default_plugin_type));
+  }
+
+  // Create graph file loader plugin
+  try {
+    plugin_type_ = nav2_util::get_plugin_type_param(node, graph_file_loader_id);
+    GraphFileLoader::Ptr graph_parser = plugin_loader_.createSharedInstance((plugin_type_));
+    RCLCPP_INFO(
+      logger_, "Created GraphFileLoader %s of type %s",
+      graph_file_loader_id.c_str(), plugin_type_.c_str());
+    graph_file_loader_ = std::move(graph_parser);
+    graph_file_loader_->configure(node);
+  } catch (pluginlib::PluginlibException & ex) {
+    RCLCPP_FATAL(
+      logger_,
+      "Failed to create GraphFileLoader. Exception: %s", ex.what());
+    throw ex;
+  }
+}
+
+bool GraphLoader::loadGraphFromFile(
+  Graph & graph,
+  GraphToIDMap & graph_to_id_map,
+  std::string filepath)
+{
+  if (filepath.empty()) {
+    RCLCPP_DEBUG(
+      logger_, "The graph filepath was not provided. "
+      "Setting to %s", graph_filepath_.c_str());
+    filepath = graph_filepath_;
+  }
+
+  RCLCPP_INFO(
+    logger_,
+    "Loading graph file from %s, by parser %s", filepath.c_str(), plugin_type_.c_str());
+
+  if (!graph_file_loader_->loadGraphFromFile(graph, graph_to_id_map, filepath)) {
+    return false;
+  }
+
+  if (!transformGraph(graph)) {
+    return false;
+  }
+
+  return true;
+}
+
+bool GraphLoader::transformGraph(Graph & graph)
+{
+  for (auto & node : graph) {
+    std::string node_frame = node.coords.frame_id;
+    if (node_frame.empty() || node_frame == route_frame_) {continue;}
+
+    if (cashed_transforms_.find(node_frame) == cashed_transforms_.end()) {
```

Comment:
`cached` is misspelled. Also, this doesn't need to be a class member, it can be constructed here and gone out of scope in case the transformations are different on different file loadings. We except loading a new file to take a little bit of time since its not a steady-state change so we can reconstruct it each time OK